kube-runtime/src/utils/predicate.rs

```diff
@@ -114,6 +115,33 @@ where
     }
 }
 
+/// Configuration for predicate filtering with cache TTL
+#[derive(Debug, Clone)]
+pub struct Config {
+    /// Time-to-live for cache entries
+    ///
+    /// Entries older than this duration will be evicted from the cache.
+    /// Default is 1 hour.
```

Comment:
```suggestion
    /// Entries not seen for during this duration will be evicted from the cache.
    /// Default is 1 hour.
```

clarifying comment because we are updating the timestamp when we re-encounter existing elements.